mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -45,16 +47,21 @@ class DatabricksArtifactRepository(ArtifactRepository):
     """
 
     def __init__(self, artifact_uri):
-        super(DatabricksArtifactRepository, self).__init__(artifact_uri)
         if not artifact_uri.startswith('dbfs:/'):
             raise MlflowException(message='DatabricksArtifactRepository URI must start with dbfs:/',
                                   error_code=INVALID_PARAMETER_VALUE)
         if not is_databricks_acled_artifacts_uri(artifact_uri):
             raise MlflowException(message=('Artifact URI incorrect. Expected path prefix to be'
                                            ' databricks/mlflow-tracking/path/to/artifact/..'),
                                   error_code=INVALID_PARAMETER_VALUE)
-        self.run_id = self._extract_run_id(self.artifact_uri)
+        # The dbfs:/ path ultimately used for artifact operations should not contain the
+        # Databricks profile info.
+        artifact_uri_no_db_profile = remove_databricks_profile_info_from_artifact_uri(artifact_uri)
+        super(DatabricksArtifactRepository, self).__init__(artifact_uri_no_db_profile)
+        self.databricks_profile_uri = get_databricks_profile_uri_from_artifact_uri(artifact_uri) \
+            or mlflow.tracking.get_tracking_uri()
```

Comment:
```suggestion
        self.databricks_profile_uri = get_databricks_profile_uri_from_artifact_uri(artifact_uri) \
            or mlflow.tracking.get_tracking_uri()
        super(DatabricksArtifactRepository, self).__init__(remove_databricks_profile_info_from_artifact_uri(artifact_uri))
```

Reply:
It seems like this would have the same impact, but it feels easier to reason about, since the databricks info is extracted and then not used again

Reply:
ok that makes sense. i'd still keep the `super` call before setting anything class variables (as a best practice to make sure `super` does not overwrite anything this class wants to overwrite).